### PR TITLE
Adaptive filter refactor

### PR DESCRIPTION
**Pull request by uvvpavel**
covers  #356

### CODE REVIEW

File: modules/lib_fdaf/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    PRIVATE
+        src/fdaf_filter.c
+        src/fdaf_control.c
+        src/fdaf_l2_filter.c
+)
+
+target_include_directories(fwk_voice_module_lib_fdaf
+    PUBLIC
+        api
+        src
```
Comment (shuchitak) [inline modules/lib_fdaf/CMakeLists.txt:15]: Why does src needs to be included?

File: modules/lib_ic/src/ic_low_level.c [hunk 2]
```diff
@@ -122,8 +118,15 @@ void ic_fft(
         bfp_complex_s32_t *output,
         bfp_s32_t *input){
 
-    aec_forward_fft(output, input);
-
+    //Input bfp_s32_t structure will get overwritten since FFT is computed in-place. Keep a copy of input->length and assign it back after fft call.
```
Comment (shuchitak) [inline modules/lib_ic/src/ic_low_level.c:121]: Can we not make the forward and inverse FFTs as part of the fdaf library?

**Review by shuchitak — COMMENTED** [review]
I've added a couple of questions as comments but otherwise looks good to me. Can we ensure that the output is bit-exact with and without these changes? For instance, run the pipeline example with and without these changes and ensure bit-exactness.

**Comment by mbanth** [comment]
A review by @andrewdewhurst and myself has concluded that we will not continue with this pull request.

**Pull request closed without merging**